services/paymaster/src/rpc.rs

```diff
@@ -0,0 +1,74 @@
+use axum::{http::StatusCode, response::ErrorResponse};
+use solana_client::{rpc_client::RpcClient, rpc_config::RpcSendTransactionConfig};
+use solana_commitment_config::CommitmentConfig;
+use solana_rpc_client_api::client_error::Error;
+use solana_transaction::versioned::VersionedTransaction;
+use solana_transaction_error::TransactionError;
+use std::time::Duration;
+use tokio::time::sleep;
+
+const GET_STATUS_RETRIES: usize = usize::MAX;
+
+#[derive(Debug, Clone, serde::Serialize)]
+#[serde(tag = "type")]
+pub enum ConfirmationResult {
+    #[serde(rename = "success")]
+    Success { signature: String },
+    #[serde(rename = "failed")]
+    Failed {
+        signature: String,
+        error: TransactionError,
+    },
+}
+
+fn to_error_response(err: Error) -> ErrorResponse {
+    (
+        StatusCode::BAD_GATEWAY,
+        format!("Failed SVM RPC call: {err}"),
+    )
+        .into()
+}
+
+// Inspired by send_and_confirm_transaction from solana-rpc-client, but accepts a config
+pub async fn send_and_confirm_transaction(
+    rpc: &RpcClient,
+    transaction: &VersionedTransaction,
+    config: RpcSendTransactionConfig,
+) -> Result<ConfirmationResult, ErrorResponse> {
+    let recent_blockhash = transaction.message.recent_blockhash();
+    let signature = rpc
+        .send_transaction_with_config(transaction, config)
+        .map_err(to_error_response)?;
+
+    for status_retry in 0..GET_STATUS_RETRIES {
```

Comment:
Is it possible to stream?

Reply:
Yes it is, we might want to revisit it later.